### DISCUSSION TITLE
Size argument of MolToImage() scales Atom Labels and Bond widths inconsistently for different sizes?

### DISCUSSION THREAD

**Discussion by p3rAsperaAdAstra** [rdkit/rdkit#6634] [Q&A]
I've already made a post on [StackOverflow](https://stackoverflow.com/questions/76866139/rdkit-image-moltoimage-scales-bond-widths-and-element-labels-inconsistently-for) about this issue, but I haven't received an answer on it yet.

The post goes into detail already, but here's the synopsis: If you generate a raster image using `MolToImage(size=(X,Y))` and you make  the size progressively bigger, the Atom Labels and Bonds are not properly scaled for the new size. The bigger the image gets, the smaller the Atom Labels are relative to the bond lengths. The bond widths are also getting thinner and thinner relative to the bond lengths the bigger `size` becomes. 

To test this I wrote some code that creates a bunch of images of Tramadol at different sizes from a SMILES string. Then I rescaled every image to 600,600 pixels and turned them into a GIF. This is the result:
![BwNVK](https://github.com/rdkit/rdkit/assets/60529579/85b7ce68-e92a-4127-a67a-4c5bf40776fe)

Here's the code I used to make this:

```python
from glob import glob
from rdkit import Chem
from rdkit.Chem import Draw
from PIL import Image,ImageDraw,ImageFont


def make_frames_from_smi(smi):

    mol = Chem.MolFromSmiles(smi)

    for i in range(10):
        s = (i+3)*100
        mol = Chem.MolFromSmiles(smi)
        img = Draw.MolToImage(mol,size=(s,s))
        img = img.resize((600,600))
        draw = ImageDraw.Draw(img)

        text = '%d: Initial Size: (%d,%d)'%(i+1,s,s)

        font_size = 40
        font = ImageFont.truetype("arial.ttf", font_size)  # Use your desired font


        # Calculate text position
        image_width, image_height = img.size
        text_x = (image_width - (bbox[2] - bbox[0])) // 2
        text_y = 20  # Adjust the vertical position as needed

        draw.text((text_x, text_y), text, font=font, fill='black')
        img.save('%03dtest.png'%i)


def make_gif_from_frames(paths):

    frames_paths = glob(paths)
    frames = [Image.open(imgp) for imgp in frames_paths]

    frames[0].save("mols.gif", format="GIF", append_images=frames, save_all=True, duration=500, loop=False)



# make RDKit mol obj.
smi = 'CN(C)CC1CCCCC1(C2=CC(=CC=C2)OC)O'
make_frames_from_smi(smi)
make_gif_from_frames('*.png')
```

Is this supposed to work like that?

**Comment by pippo1990** [comment]
https://stackoverflow.com/questions/76866139/rdkit-image-moltoimage-scales-bond-widths-and-element-labels-inconsistently-for/79012036#79012036

**Comment by greglandrum** [comment]
Hi @pippo1990 
The RDKit by default does not change the bond width when the image size changes. You can change this or the bond width that is used with the drawing options.

The font size is only allowed to vary within a particular range. This is also under your control using drawing options.

Here's a quick demo of what you might do:
![image](https://github.com/user-attachments/assets/07931da5-7ebe-412a-a065-24ae2053095d)


More information on the available drawing parameters is here:
https://greglandrum.github.io/rdkit-blog/posts/2023-05-26-drawing-options-explained.html

**Reply by pippo1990** [reply]
Thanks , used dopts.bondLineWidth , in first answer( #default is 2.0 to me, not sure about it)  , didnt encouter scaleBondWidth so fa in my googling efforts. Thanks again
